### ISSUE TITLE
TestActionsService_DownloadArtifact_invalidLocationHeader fails on Go1.20

### ISSUE THREAD

**Issue by zhsj** [google/go-github#2626]
Tested on master version.

```
$ go version
go version go1.20rc3 linux/amd64
```

```
$ go test -run TestActionsService_DownloadArtifact_invalidLocationHeader
--- FAIL: TestActionsService_DownloadArtifact_invalidLocationHeader (0.00s)
    actions_artifacts_test.go:385: Expected URL parse error, got <nil>
FAIL
exit status 1
FAIL    github.com/google/go-github/v49/github  0.008s
```

**Comment by gmlewis** [comment]
Thank you, @zhsj . Confirmed.

That seems odd. Whoever wants to take on this issue should read through the go1.20rc3 release notes and/or changelist and see what could be causing this, then find out if it is a bug in the release candidate or a bug in this repo or a breaking change.

**Comment by gmlewis** [comment]
Just for the record, I get two failures:

```bash
$ go test ./...
--- FAIL: TestActionsService_DownloadArtifact_invalidLocationHeader (0.00s)
    actions_artifacts_test.go:385: Expected URL parse error, got <nil>
--- FAIL: TestRepositoriesService_GetArchiveLink_invalidLocationHeader (0.00s)
    repos_contents_test.go:761: Expected URL parse error, got <nil>
FAIL
FAIL	github.com/google/go-github/v49/github	0.360s
FAIL
```

**Comment by gofri** [comment]
I updated `testURLParseError()` as follows:
```
-       if err, ok := err.(*url.Error); !ok || err.Op != "parse" {
-               t.Errorf("Expected URL parse error, got %+v", err)
+       if nerr, ok := err.(*url.Error); !ok || nerr.Op != "parse" {
+               t.Errorf("Expected URL parse error, got %v %+v (%v->%T)", ok, nerr, err, errors.Unwrap(err))
```
And got:
```
--- FAIL: TestActionsService_DownloadArtifact_invalidLocationHeader (0.00s)
    actions_artifacts_test.go:385: Expected URL parse error, got false <nil> (net/http: HTTP/1.x transport connection broken: malformed MIME header line: Location: https://google.com->textproto.ProtocolError)
```

IMO this test is still helpful (there is a real error), but it needs to handle this new error type.
also, it might be a good idea to add the printing of the ok value there (and maybe also the original error too; the Unwrap() one is unsafe if ==nil)

**Comment by gmlewis** [comment]
Whups, I forgot about this issue when I made #2654 and resolved the issue in #2656.

I dug into these tests, and believe they are redundant.  Yes, there is a real error, but it is caught in a different place and the test was written to exercise a specific code path which is impossible to cover with Go 1.20.

I believe this issue can be closed if there are no objections.

**Comment by zhsj** [comment]
Yeah, the test is removed by https://github.com/google/go-github/pull/2656
